api/apps/v1alpha1/nemo_guardrails_types.go

```diff
@@ -81,6 +80,26 @@ type NemoGuardrailSpec struct {
 	RuntimeClass string `json:"runtimeClass,omitempty"`
 }
 
+type NIMEndpoint struct {
+	// The base URL for the NIM service. This can either be the endpoint for a single NIM or a NIM proxy.
+	// A NIM proxy endpoint is needed if you need to run guardrail for serving multiple NIMs.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Pattern=`^https?:\/\/[^\s]+\/v1\/?$`
```

Comment:
This `https` check will block locally hosted NIMs served over http. We should allow both for flexibility.